lib/elixir/src/elixir_expand.erl

```diff
@@ -776,23 +776,34 @@ check_erlang_operator_args({{'.', _, [erlang, '++']}, Meta, [ELeft, _]}, [Left,
     false -> form_error(Meta, ?key(E, file), ?MODULE, {invalid_arg_for_lists_concatenation, Left});
     true -> ok
   end;
-check_erlang_operator_args({{'.', _, [erlang, Op]}, Meta, [ELeft, ERight]}, [Left, Right], _, E)
+check_erlang_operator_args({{'.', _, [erlang, Op]}, Meta, [ELeft, ERight]} = Node, [Left, Right], _, E)
     when Op =:= '>'; Op =:= '<'; Op =:= '=<'; Op =:= '>=' ->
+
   Result =
     case is_struct_expression(ELeft) of
-      true -> Left;
+      true -> {Left, nil};
       false ->
         case is_struct_expression(ERight) of
-          true -> Right;
-          false -> false
+          true -> {Right, nil};
+          false ->
+            case is_comparison_expression(Left) of
```

Comment:
We should not run validations on `Left` and `Right` because they haven't been expanded yet, so we may have false positives in case someone overrides `<` and friends. So you need to run this on `ELeft` and `ERight`.

Reply:
:+1: